src/ui/utils/redux.ts

```diff
@@ -0,0 +1,34 @@
+import { parseSync, traverse } from "@babel/core";
+import { Location } from "@replayio/protocol";
+
+export function isReduxMiddleware(sourceContents: string, location: Location) {
+  try {
+    const ast = parseSync(sourceContents, {});
+    let isMiddleware = false;
+
+    if (ast) {
+      traverse(ast, {
+        Identifier(path) {
+          const node = path.node;
+          const loc = node.loc;
+          if (loc?.start.line === location.line && loc.start.column === location.column) {
+            // found the node that we paused at
+            // so the parent callback here should be of type `store => next => action => {}`
+            const dispatch = path.getFunctionParent();
+            const wrapDispatch = dispatch?.getFunctionParent();
+            const middleware = wrapDispatch?.getFunctionParent();
+
+            if ([dispatch, wrapDispatch, middleware].every(f => f?.node.params.length === 1)) {
+              isMiddleware = true;
+              path.stop();
+            }
+          }
+        },
+      });
+    }
+
+    return isMiddleware;
+  } catch (e) {
+    return false;
+  }
+}
```

Comment:
it'd be nice to write some jest unit tests for this function

https://github.com/replayio/devtools/blob/7731de4856591274a04b4596dd22d37416ec34f6/src/ui/utils/time.test.ts#L4
https://github.com/replayio/devtools/blob/7731de4856591274a04b4596dd22d37416ec34f6/src/ui/utils/app.test.ts#L4